src/android/ContactAccessorSdk5.java

```diff
@@ -307,24 +306,37 @@ public JSONArray search(JSONArray fields, JSONObject options) {
      * @return     a JSONObject representing the contact
      * @throws JSONException
      */
+    @Override
     public JSONObject getContactById(String id) throws JSONException {
-        // Call overloaded version with no desiredFields
-        return getContactById(id, null);
+        return getContactById(ContactsContract.Data.CONTACT_ID, id);
     }
 
+    /**
+     * A special search that finds one contact by its raw id
+     *
+     * @param rawId   contact to find by raw id
+     * @return     a JSONObject representing the contact
+     * @throws JSONException
+     */
     @Override
-    public JSONObject getContactById(String id, JSONArray desiredFields) throws JSONException {
-        // Do the id query
+    public JSONObject getContactByRawId(String rawId) throws JSONException {
+        return getContactById(ContactsContract.Data.RAW_CONTACT_ID, rawId);
+    }
+
+    private JSONObject getContactById(String idKey, String idValue) throws JSONException {
```

Comment:
In actuality, now that I'm seeing again, `idKey` is a bad name for it. It should be more like `idColumnName` and the other `idValue` or just `id`. It's not a key-value pair per se, since it isn't a map/dictionary, but rather a DB.

Does that make sense?

Reply:
I ended up doing that change - 2ad91bd1b6a4bc3f4856fcd62ea94fa99a99a233 - if you'd like to take another look.

Reply:
It does. Still, even if the relationship is not a key-value pair per se, both arguments relate to each other. That's why I see value in grouping them. A structure/class for this specific use case seems too much so I suggested the Pair structure.

Reply:
Understood, The `Pair` class is kotlin specific I think, but I'll use something similar in Java.

Reply:
1b59b5055e9393760e707d2fc3070f5e09430025